more stores on AWS

tbg opened the pull request:
work towards #3471:

`terraform apply <usual_stuff> --var=aux_data_dirs=data2,data3`

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/3566)

<!-- Reviewable:end -->

mberhault commented on cloud/aws/variables.tf:
```diff
@@ -6,6 +6,11 @@ variable "cockroach_port" {
 
+# Comma-separated list of directories added as auxiliary stores.
+variable "aux_data_dirs" {
+  default = ""
```
there should really be a default here.

tschottdorf replied:
there's a default. It's an empty string, that means you get `data` and nothing else. I didn't want `data` itself to be configurable because that seemed like a hassle.

mberhault replied:
yeah, that's weird. I would make the default `"data"`, and keep it out of the `formatlist`

tschottdorf replied:
I'm not convinced what for, all that gives us is more ways of screwing things up and more to check. There always needs to be at least one store. What's the part about `formatlist` about?

mberhault replied:
because you're assuming that `data` will always be a store. When we get to the point of using multiple real stores per node (soonish probably, I have my eye on a few multi-SSD instances), we'll need to set all the paths, not just the ones other than data.

tschottdorf replied:
ok. I'll rename the variable to `stores` and make sure we don't hard-code `data` anywhere.

mberhault commented:
LGTM

tbg committed:
more stores on AWS

work towards #3471:

terraform apply <usual_stuff> --var=stores=ssd=/tmp/data1,ssd=/tmp/data2

tschottdorf commented:
done, PTAL

mberhault commented on cloud/aws/main.tf:
```diff
@@ -83,7 +84,7 @@ resource "null_resource" "cockroach-initializer" {
       "bash download_binary.sh cockroach/cockroach ${var.cockroach_sha}",
       "mkdir -p logs",
-      "./cockroach init --logtostderr=true --stores=sdd=data",
+      "./cockroach init --logtostderr=true --stores=${element(split(",", "${var.stores}"), 0)}",
```
why `element(..., 0)`? if you provide `--var=stores=ssd=data,hdd=data2`, it will just use `ssd=data`.
You should also make sure that the extra `=` doesn't cause issues (eg: if they just call `strings.Split`, we're hosed)

tschottdorf replied:
which extra `=`? Removed the `element(..., 0)`.

tschottdorf replied:
Ok, this will actually try to bootstrap both stores which dies on start, so we need the `element(..., 0)`. Waiting for explanation of your `=` comment before merge (and maybe we should open an issue about the bootstrapping process).

mberhault replied:
The flag passed to terraform will be something along the lines of: `--var=stores=ssd=data,ssd=data2`. I'm just curious whether this works, which depends on how terraform parses the `--var` flag.
If they do a `strings.Split('=')` they'll split the `ssd=data` parameter as well.
If they just consume the first `=`, then it's fine. Good to test it out though.

mberhault replied:
as for the `init` step, it can use `element(0,...)`, and the `start` step can keep all of them.

tschottdorf replied:
oh, no. I tested, it does do the right thing.

mberhault commented:
LGTM
